In [1]:
%matplotlib inline
import pandas as pd
from pandas import ExcelWriter
from pandas import ExcelFile
import matplotlib.pyplot as plt
import math
import numpy as np

import altair as alt
from vega_datasets import data


olympics = pd.read_csv('data/athlete_events.csv') 
region = pd.read_csv('data/noc_regions.csv')

In [2]:
olympics.head(10)

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
5   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   
6   5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
7   5  Christine Jacoba Aaftink   F  25.0   185.0    82.0     Netherlands   
8   5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   
9   5  Christine Jacoba Aaftink   F  27.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season         City          Sport  \
0  CHN  1992 Summer  1992  Summer    Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer       London           Judo   
2  DEN  1920 Summer  1920  Summer    Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer        Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
5  NED  1988 Winter  1988  Winter      Calgary  Speed Skating   
6  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
7  NED  1992 Winter  1992  Winter  Albertville  Speed Skating   
8  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   
9  NED  1994 Winter  1994  Winter  Lillehammer  Speed Skating   

                                Event Medal  
0         Basketball Men's Basketball   NaN  
1        Judo Men's Extra-Lightweight   NaN  
2             Football Men's Football   NaN  
3         Tug-Of-War Men's Tug-Of-War  Gold  
4    Speed Skating Women's 500 metres   NaN  
5  Speed Skating Women's 1,000 metres   NaN  
6    Speed Skating Women's 500 metres   NaN  
7  Speed Skating Women's 1,000 metres   NaN  
8    Speed Skating Women's 500 metres   NaN  
9  Speed Skating Women's 1,000 metres   NaN

# Comparing the Most Recent Olympics to the First

What was the first year recorded? 

In [3]:
olympics.min()['Year']

1896

Make a subset of the data of just the 1896 data 

In [4]:
first = olympics["Year"] == 1896
athina96 = olympics[first]
athina96.head(10)

ID                                Name Sex  Age  Height  Weight  \
3079   1724              Aristidis Akratopoulos   M  NaN     NaN     NaN   
3080   1724              Aristidis Akratopoulos   M  NaN     NaN     NaN   
3081   1725  Konstantinos "Kostas" Akratopoulos   M  NaN     NaN     NaN   
3082   1725  Konstantinos "Kostas" Akratopoulos   M  NaN     NaN     NaN   
7348   4113                  Anastasios Andreou   M  NaN     NaN     NaN   
7353   4116                     Ioannis Andreou   M  NaN     NaN     NaN   
7498   4189             Nikolaos Andriakopoulos   M  NaN     NaN     NaN   
7998   4431                    Georgios Anninos   M  NaN     NaN     NaN   
8144   4493                      Antelothanasis   M  NaN     NaN     NaN   
10332  5660                   Georgios Aspiotis   M  NaN     NaN     NaN   

           Team  NOC        Games  Year  Season    City       Sport  \
3079     Greece  GRE  1896 Summer  1896  Summer  Athina      Tennis   
3080   Greece-3  GRE  1896 Summer  1896  Summer  Athina      Tennis   
3081     Greece  GRE  1896 Summer  1896  Summer  Athina      Tennis   
3082   Greece-3  GRE  1896 Summer  1896  Summer  Athina      Tennis   
7348     Greece  GRE  1896 Summer  1896  Summer  Athina   Athletics   
7353     Greece  GRE  1896 Summer  1896  Summer  Athina    Swimming   
7498     Greece  GRE  1896 Summer  1896  Summer  Athina  Gymnastics   
7998     Greece  GRE  1896 Summer  1896  Summer  Athina    Swimming   
8144     Greece  GRE  1896 Summer  1896  Summer  Athina    Shooting   
10332    Greece  GRE  1896 Summer  1896  Summer  Athina     Cycling   

                                                   Event   Medal  
3079                                Tennis Men's Singles     NaN  
3080                                Tennis Men's Doubles     NaN  
3081                                Tennis Men's Singles     NaN  
3082                                Tennis Men's Doubles     NaN  
7348                  Athletics Men's 110 metres Hurdles     NaN  
7353               Swimming Men's 1,200 metres Freestyle  Silver  
7498                      Gymnastics Men's Rope Climbing    Gold  
7998                 Swimming Men's 100 metres Freestyle     NaN  
8144   Shooting Men's Free Rifle, Three Positions, 30...     NaN  
10332                Cycling Men's Road Race, Individual     NaN

What is the most recent games? 

In [5]:
olympics.max()['Year']

2016

Make a subset of the data of just 2016 data. 

In [6]:
last = olympics["Year"] == 2016
rio16 = olympics[last]
rio16.head(10)

ID                 Name Sex   Age  Height  Weight        Team  NOC  \
80   22        Andreea Aanei   F  22.0   170.0   125.0     Romania  ROU   
139  51    Nstor Abad Sanjun   M  23.0   167.0    64.0       Spain  ESP   
140  51    Nstor Abad Sanjun   M  23.0   167.0    64.0       Spain  ESP   
141  51    Nstor Abad Sanjun   M  23.0   167.0    64.0       Spain  ESP   
142  51    Nstor Abad Sanjun   M  23.0   167.0    64.0       Spain  ESP   
143  51    Nstor Abad Sanjun   M  23.0   167.0    64.0       Spain  ESP   
144  51    Nstor Abad Sanjun   M  23.0   167.0    64.0       Spain  ESP   
148  55  Antonio Abadia Beci   M  26.0   170.0    65.0       Spain  ESP   
158  62    Giovanni Abagnale   M  21.0   198.0    90.0       Italy  ITA   
161  65    Patimat Abakarova   F  21.0   165.0    49.0  Azerbaijan  AZE   

           Games  Year  Season            City          Sport  \
80   2016 Summer  2016  Summer  Rio de Janeiro  Weightlifting   
139  2016 Summer  2016  Summer  Rio de Janeiro     Gymnastics   
140  2016 Summer  2016  Summer  Rio de Janeiro     Gymnastics   
141  2016 Summer  2016  Summer  Rio de Janeiro     Gymnastics   
142  2016 Summer  2016  Summer  Rio de Janeiro     Gymnastics   
143  2016 Summer  2016  Summer  Rio de Janeiro     Gymnastics   
144  2016 Summer  2016  Summer  Rio de Janeiro     Gymnastics   
148  2016 Summer  2016  Summer  Rio de Janeiro      Athletics   
158  2016 Summer  2016  Summer  Rio de Janeiro         Rowing   
161  2016 Summer  2016  Summer  Rio de Janeiro      Taekwondo   

                                       Event   Medal  
80   Weightlifting Women's Super-Heavyweight     NaN  
139   Gymnastics Men's Individual All-Around     NaN  
140          Gymnastics Men's Floor Exercise     NaN  
141           Gymnastics Men's Parallel Bars     NaN  
142          Gymnastics Men's Horizontal Bar     NaN  
143                   Gymnastics Men's Rings     NaN  
144         Gymnastics Men's Pommelled Horse     NaN  
148             Athletics Men's 5,000 metres     NaN  
158               Rowing Men's Coxless Pairs  Bronze  
161              Taekwondo Women's Flyweight  Bronze

Now that I have created two subsets for the 1896 and 2016 Olympics I will now try to answer a series of questions to compare the difference between the Olympics when it first came to be, compared to what it is today.

**1) Which countries participated in each respective Olympics** 

In [7]:
athina96["NOC"].nunique()

12

12 different countries participated in the 1896 Summer Games

In [8]:
rio16["NOC"].nunique()

207

207 different countries participated in the 2016 Summer Games 

*Maybe merge the region dataset in ??*

**Which countries medaled most?**

In [9]:
medal_count96 = athina96.groupby(['NOC'])["Medal"].value_counts()

In [10]:
print(medal_count96)

NOC  Medal 
AUS  Gold       2
     Bronze     1
AUT  Bronze     2
     Gold       2
     Silver     1
DEN  Bronze     3
     Silver     2
     Gold       1
FRA  Gold       5
     Silver     4
     Bronze     2
GBR  Bronze     3
     Gold       3
     Silver     3
GER  Gold      25
     Silver     5
     Bronze     2
GRE  Bronze    20
     Silver    18
     Gold      10
HUN  Bronze     3
     Gold       2
     Silver     1
SUI  Silver     2
     Gold       1
USA  Gold      11
     Silver     7
     Bronze     2
Name: Medal, dtype: int64


In [11]:
total_medal96 = medal_count96.groupby(['NOC']).sum()

In [12]:
print(total_medal96)

NOC
AUS     3
AUT     5
DEN     6
FRA    11
GBR     9
GER    32
GRE    48
HUN     6
SUI     3
USA    20
Name: Medal, dtype: int64


Greece medaled the most in the 1896 games with 48 total medals. 

**DO THE SAME FOR 2016**

In [13]:
medal_count16 = rio16.groupby(['NOC'])["Medal"].value_counts()

In [14]:
total_medal16 = medal_count16.groupby(['NOC']).sum()
print(total_medal16)

NOC
ALG      2
ARG     22
ARM      4
AUS     82
AUT      2
AZE     18
BAH      6
BDI      1
BEL     21
BLR     12
BRA     50
BRN      2
BUL      7
CAN     69
CHN    113
CIV      2
COL      8
CRO     24
CUB     11
CZE     15
DEN     41
DOM      1
EGY      3
ESP     45
EST      4
ETH      8
FIJ     13
FIN      1
FRA     96
GBR    145
      ... 
NIG      1
NOR     19
NZL     36
PHI      1
POL     16
POR      1
PRK      7
PUR      1
QAT      1
ROU     16
RSA     23
RUS    115
SGP      1
SLO      4
SRB     54
SUI     11
SVK      8
SWE     28
THA      6
TJK      1
TPE      5
TTO      1
TUN      3
TUR      8
UAE      1
UKR     15
USA    264
UZB     13
VEN      3
VIE      2
Name: Medal, Length: 86, dtype: int64


In [15]:
total_medal16.max()

264

USA medaled the most in 2016 games with 264 total medals. 

## Which Countries Won the Most Gold Medals? 

#### 1896

In [16]:
gold = athina96['Medal'] == "Gold"
gold96 = athina96[gold]

In [17]:
total_gold96 = gold96.groupby(["NOC"])["Medal"].value_counts()
print(total_gold96)

NOC  Medal
AUS  Gold      2
AUT  Gold      2
DEN  Gold      1
FRA  Gold      5
GBR  Gold      3
GER  Gold     25
GRE  Gold     10
HUN  Gold      2
SUI  Gold      1
USA  Gold     11
Name: Medal, dtype: int64


Greece has the most gold medals with 25 in the 1896 Olympic Games 

#### 2016

In [18]:
gold2 = rio16['Medal'] == "Gold"
gold16 = rio16[gold2]

In [19]:
total_gold16 = gold16.groupby(["NOC"])["Medal"].value_counts()
print(total_gold16)

NOC  Medal
ARG  Gold      21
ARM  Gold       1
AUS  Gold      23
AZE  Gold       1
BAH  Gold       1
BEL  Gold       2
BLR  Gold       1
BRA  Gold      36
BRN  Gold       1
CAN  Gold       4
CHN  Gold      46
CIV  Gold       1
COL  Gold       3
CRO  Gold       7
CUB  Gold       5
CZE  Gold       1
DEN  Gold      15
ESP  Gold       9
ETH  Gold       1
FIJ  Gold      13
FRA  Gold      20
GBR  Gold      64
GEO  Gold       2
GER  Gold      49
GRE  Gold       3
HUN  Gold      12
INA  Gold       2
IOA  Gold       1
IRI  Gold       3
ITA  Gold       8
JAM  Gold      11
JOR  Gold       1
JPN  Gold      17
KAZ  Gold       3
KEN  Gold       6
KOR  Gold      13
KOS  Gold       1
NED  Gold       9
NZL  Gold       6
POL  Gold       3
PRK  Gold       2
PUR  Gold       1
ROU  Gold       4
RSA  Gold       2
RUS  Gold      52
SGP  Gold       1
SLO  Gold       1
SRB  Gold      14
SUI  Gold       6
SVK  Gold       3
SWE  Gold       2
THA  Gold       2
TJK  Gold       1
TPE  Gold       1
TUR  Gold       1

In [20]:
total_gold16.max()

139

USA won the most gold medals with 139 in the 2016 Olympic Games. 

## Looking at Sports Across the Years

**How many sports were there in each game?**

#### 1896

In [21]:
athina96['Sport'].nunique()

9

There were only 9 different sports in the 1896 Olympics 

In [22]:
athina96['Sport'].unique()

array(['Tennis', 'Athletics', 'Swimming', 'Gymnastics', 'Shooting',
       'Cycling', 'Fencing', 'Weightlifting', 'Wrestling'], dtype=object)

The sports played in the 1896 Olympics included Tennis, Swimming, Gymnastics, Shooting, Cycling, Fencing, Weightlifting and Wrestling

#### 2016

In [23]:
rio16['Sport'].nunique()

34

There were 34 sports played in the 2016 Olympics. 

In [24]:
rio16['Sport'].unique()

array(['Weightlifting', 'Gymnastics', 'Athletics', 'Rowing', 'Taekwondo',
       'Handball', 'Boxing', 'Equestrianism', 'Cycling', 'Volleyball',
       'Diving', 'Shooting', 'Judo', 'Football', 'Synchronized Swimming',
       'Wrestling', 'Fencing', 'Swimming', 'Badminton', 'Rugby Sevens',
       'Table Tennis', 'Water Polo', 'Trampolining', 'Basketball',
       'Triathlon', 'Modern Pentathlon', 'Sailing', 'Beach Volleyball',
       'Golf', 'Rhythmic Gymnastics', 'Archery', 'Hockey', 'Canoeing',
       'Tennis'], dtype=object)

## Looking at Athletes  

**How many athletes were there in each game?**

#### 1896

In [25]:
athina96.head()

ID                                Name Sex  Age  Height  Weight  \
3079  1724              Aristidis Akratopoulos   M  NaN     NaN     NaN   
3080  1724              Aristidis Akratopoulos   M  NaN     NaN     NaN   
3081  1725  Konstantinos "Kostas" Akratopoulos   M  NaN     NaN     NaN   
3082  1725  Konstantinos "Kostas" Akratopoulos   M  NaN     NaN     NaN   
7348  4113                  Anastasios Andreou   M  NaN     NaN     NaN   

          Team  NOC        Games  Year  Season    City      Sport  \
3079    Greece  GRE  1896 Summer  1896  Summer  Athina     Tennis   
3080  Greece-3  GRE  1896 Summer  1896  Summer  Athina     Tennis   
3081    Greece  GRE  1896 Summer  1896  Summer  Athina     Tennis   
3082  Greece-3  GRE  1896 Summer  1896  Summer  Athina     Tennis   
7348    Greece  GRE  1896 Summer  1896  Summer  Athina  Athletics   

                                   Event Medal  
3079                Tennis Men's Singles   NaN  
3080                Tennis Men's Doubles   NaN  
3081                Tennis Men's Singles   NaN  
3082                Tennis Men's Doubles   NaN  
7348  Athletics Men's 110 metres Hurdles   NaN

In [26]:
athina96["ID"].count()

380

There were only 380 athletes in the 1896 games 

#### 2016

In [27]:
rio16["ID"].count()

13688

There were 13,688 athletes in the 2016 Olympics. 

**What was the average age of athletes?**

#### 1896

In [28]:
athina96["Age"].min()

10.0

The youngest athlete was 10

In [29]:
athina96[athina96["Age"]==10]

ID                Name Sex   Age  Height  Weight  \
142882  71691  Dimitrios Loundras   M  10.0     NaN     NaN   

                                 Team  NOC        Games  Year  Season    City  \
142882  Ethnikos Gymnastikos Syllogos  GRE  1896 Summer  1896  Summer  Athina   

             Sport                                  Event   Medal  
142882  Gymnastics  Gymnastics Men's Parallel Bars, Teams  Bronze

In [30]:
athina96["Age"].max()

40.0

The oldest athlete was 40

In [31]:
athina96[athina96["Age"]==40]

ID                          Name Sex   Age  Height  Weight  \
255783  128057  Charles Waldstein (-Walston)   M  40.0     NaN     NaN   

                 Team  NOC        Games  Year  Season    City     Sport  \
255783  United States  USA  1896 Summer  1896  Summer  Athina  Shooting   

                                            Event Medal  
255783  Shooting Men's Military Rifle, 200 metres   NaN

In [32]:
athina96["Age"].mean()

23.580645161290324

The average age of the athletes in 1896 was 23.58

#### 2016 

In [33]:
rio16["Age"].min()

13.0

The youngest athlete in 2016 was 13

In [34]:
rio16[rio16["Age"]==13]

ID             Name Sex   Age  Height  Weight     Team  NOC  \
49377    25434  Ana Iulia Dascl   F  13.0   183.0    60.0  Romania  ROU   
221023  111044    Gaurika Singh   F  13.0   155.0    45.0    Nepal  NEP   

              Games  Year  Season            City     Sport  \
49377   2016 Summer  2016  Summer  Rio de Janeiro  Swimming   
221023  2016 Summer  2016  Summer  Rio de Janeiro  Swimming   

                                         Event Medal  
49377    Swimming Women's 100 metres Freestyle   NaN  
221023  Swimming Women's 100 metres Backstroke   NaN

In [35]:
rio16["Age"].max()

62.0

The oldest athlete was 62 

In [36]:
rio16[rio16["Age"]==62]

ID            Name Sex   Age  Height  Weight         Team  NOC  \
29912  15436  Julie Brougham   F  62.0   157.0    48.0  New Zealand  NZL   

             Games  Year  Season            City          Sport  \
29912  2016 Summer  2016  Summer  Rio de Janeiro  Equestrianism   

                                          Event Medal  
29912  Equestrianism Mixed Dressage, Individual   NaN

In [37]:
rio16["Age"].mean()

26.20791934541204

The average age of athletes in 2016 was 26.21 

## Which athlete won the most medals

#### 1896

In [38]:
athlete_medal_count96 = athina96.groupby(['ID'])["Medal"].value_counts()
total_athlete_medal_count96 = athlete_medal_count96.groupby(['ID']).sum()

In [39]:
print(total_athlete_medal_count96)

ID
4116      1
4189      1
8620      1
12068     1
12563     2
12929     2
16616     2
17492     1
18785     2
21795     2
22700     3
24423     1
25072     1
25184     1
29467     1
30049     1
32330     2
35698     3
35708     3
35740     4
35741     2
36739     3
38860     4
39588     1
40851     1
40985     1
41160     1
42069     1
42102     1
42499     1
         ..
91250     1
91355     1
91932     1
92301     1
93091     1
93575     1
93606     1
93607     1
94075     1
94715     1
97712     1
101352    1
102806    2
107006    3
107607    2
107613    4
117571    1
118532    1
119585    1
121713    1
121920    1
122329    1
122526    1
122937    1
125390    1
126142    2
127573    1
129377    6
131985    1
135481    3
Name: Medal, Length: 93, dtype: int64


In [40]:
total_athlete_medal_count96.max()

6

In [41]:
most_medals = total_athlete_medal_count96 == 6

In [42]:
total_athlete_medal_count96[most_medals]

ID
129377    6
Name: Medal, dtype: int64

Find which athlete won the most medals 

In [43]:
athina96[athina96["ID"] == 129377]

ID                            Name Sex   Age  Height  Weight  \
258382  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   
258383  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   
258384  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   
258385  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   
258386  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   
258387  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   
258388  129377  Hermann Otto Ludwig Weingrtner   M  31.0     NaN     NaN   

           Team  NOC        Games  Year  Season    City       Sport  \
258382  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   
258383  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   
258384  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   
258385  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   
258386  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   
258387  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   
258388  Germany  GER  1896 Summer  1896  Summer  Athina  Gymnastics   

                                         Event   Medal  
258382            Gymnastics Men's Horse Vault  Bronze  
258383          Gymnastics Men's Parallel Bars     NaN  
258384   Gymnastics Men's Parallel Bars, Teams    Gold  
258385         Gymnastics Men's Horizontal Bar    Gold  
258386  Gymnastics Men's Horizontal Bar, Teams    Gold  
258387                  Gymnastics Men's Rings  Silver  
258388        Gymnastics Men's Pommelled Horse  Silver

The top medal earner was Hermann Otto Ludwig from Germany. He won 6 total medals in Gymnastics. 

### 2016

In [44]:
athlete_medal_count16 = rio16.groupby(['ID'])["Medal"].value_counts()
total_athlete_medal_count16 = athlete_medal_count16.groupby(['ID']).sum()

In [45]:
total_athlete_medal_count16.max()

6

In [46]:
most_medals = total_athlete_medal_count16 == 6
total_athlete_medal_count16[most_medals]

ID
94406    6
Name: Medal, dtype: int64

In [47]:
rio16[rio16["ID"] == 94406]

ID                     Name Sex   Age  Height  Weight  \
187911  94406  Michael Fred Phelps, II   M  31.0   193.0    91.0   
187912  94406  Michael Fred Phelps, II   M  31.0   193.0    91.0   
187913  94406  Michael Fred Phelps, II   M  31.0   193.0    91.0   
187914  94406  Michael Fred Phelps, II   M  31.0   193.0    91.0   
187915  94406  Michael Fred Phelps, II   M  31.0   193.0    91.0   
187916  94406  Michael Fred Phelps, II   M  31.0   193.0    91.0   

                 Team  NOC        Games  Year  Season            City  \
187911  United States  USA  2016 Summer  2016  Summer  Rio de Janeiro   
187912  United States  USA  2016 Summer  2016  Summer  Rio de Janeiro   
187913  United States  USA  2016 Summer  2016  Summer  Rio de Janeiro   
187914  United States  USA  2016 Summer  2016  Summer  Rio de Janeiro   
187915  United States  USA  2016 Summer  2016  Summer  Rio de Janeiro   
187916  United States  USA  2016 Summer  2016  Summer  Rio de Janeiro   

           Sport                                          Event   Medal  
187911  Swimming  Swimming Men's 4 x 100 metres Freestyle Relay    Gold  
187912  Swimming  Swimming Men's 4 x 200 metres Freestyle Relay    Gold  
187913  Swimming            Swimming Men's 100 metres Butterfly  Silver  
187914  Swimming            Swimming Men's 200 metres Butterfly    Gold  
187915  Swimming    Swimming Men's 200 metres Individual Medley    Gold  
187916  Swimming     Swimming Men's 4 x 100 metres Medley Relay    Gold

In 2016, Michael Phelps won the most medals, with 6 medals in swimming. 